pkg/api/api_test.go

```diff
@@ -0,0 +1,128 @@
+package api
+
+import (
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/rvolosatovs/turtlitto/pkg/api/apitest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStateDifference(t *testing.T) {
+	type TestCase struct {
+		inputOldState       *State
+		inputNewState       *State
+		expectedOutputState *State
+	}
+
+	for i, tc := range []TestCase{
+		//test case 1
+		{
+			inputOldState: &State{
+				Command: CommandStop,
+				Turtles: map[string]*TurtleState{
+					"1": {
+						Kinect1State:    KinectStateNoBall,
+						EmergencyStatus: apitest.Uint8Ptr(0),
+					},
+					"2": {
+						Kinect1State:    KinectStateBall,
```

Comment:
For this test case add some field, which is included in the `old` struct, but not defined in the `new`, to make sure it's not removed.